epoch.go

```diff
@@ -1747,6 +1749,16 @@ func (e *Epoch) createFinalizedBlockVerificationTask(block Block, finalization F
 		verifiedBlock, err := block.Verify(context.Background())
 		if err != nil {
 			e.Logger.Debug("Failed verifying block", zap.Error(err))
+			// if we fail to verify the block, we re-add to request timeout
+			numSigners := int64(len(finalization.QC.Signers()))
+			index, err := rand.Int(rand.Reader, big.NewInt(numSigners))
```

Comment:
I don't understand why sample a number from 0 to num signers. Don't we need to sample the number from `0` to `n-1` ?